File: static/code/sample/tethered/gdl-cmserusb-a.cs
// In this sample:
// Receive a byte
// Send back (byte + 1)

using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

void Initialize() {
    duelink.Uart.Configuration(9600,128);    
}

void WriteByte(byte b) {
    duelink.Uart.WriteByte(b);
}

byte ReadByte() {
    var data = duelink.Uart.ReadByte();
    return data;    
}

int ByteToRead() {
    var count = duelink.Uart.BytesToRead();
    return count;
}

Initialize();

while (true) {        
    while (ByteToRead() == 0) {
        Thread.Sleep(100);
    }
    var b = ReadByte();

    b++;

    WriteByte(b);           
}

File: static/code/sample/tethered/gdl-cmserusb-a.py
# In this sample:
# Receive a byte
# Send back (byte + 1)

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
def Initialize():
    duelink.Uart.Configuration(9600, 128)

def WriteByte(b):
    duelink.Uart.WriteByte(b)

def ReadByte():
    data = duelink.Uart.ReadByte()
    return data

def ByteToRead():
    count = duelink.Uart.BytesToRead()
    return count

Initialize()

while True:
    while ByteToRead() == 0:
        time.sleep(0.1)  # 100 ms, same as Thread.Sleep(100)

    b = ReadByte()
    b += 1
    WriteByte(b)